### PR TITLE
Add png-surface

### PR DESCRIPTION
**Pull request by szledan**
Signed-off-by: Szilard Ledan <szledan@gmail.com>

### CODE REVIEW

File: src/surfaces/gepard-png-surface.h [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+ * \brief A simple PNGSurface class for _Gepard_
+ *
+ * \todo: documentation is missing.
+ */
+class PNGSurface : public Surface {
+public:
+    PNGSurface(uint32_t width = 0, uint32_t height = 0, std::string fileName = "gepard.png", std::string title = "Rendered with Gepard")
+        : Surface(width, height)
+        , _buffer(malloc(width * height * sizeof(uint32_t)))
+        , _fileName(fileName)
```
Comment (dbatyai) [inline src/surfaces/gepard-png-surface.h]: Do we need to store the file name in the surface? As far as I can tell it's only used in the `save` method, where we already have it as a parameter.

File: src/gepard-engine.h [hunk 2]
```diff
@@ -84,7 +83,6 @@ class GepardEngine {
     void setFillColor(const Float red, const Float green, const Float blue, const Float alpha = 1.0f);
 
 private:
-    Surface* _surface;
```
Comment (kkristof) [inline src/gepard-engine.h]: Won't we need the _surface's width/height for the common part? If we need it, then you should add getSurfaceWidth/Height() function for the backend.

Comment (szledan) [inline src/gepard-engine.h]: You are right. I am going to refactor this functionality in another patch, but I think is bigger issue than this patch.

Comment (kkristof) [inline src/gepard-engine.h]: Okay, please fill an issue for it.

**Pull request merged**